movement/roi/line.py

```diff
@@ -56,3 +66,94 @@ def __init__(
 
         """
         super().__init__(points, dimensions=1, closed=loop, name=name)
+
+    @broadcastable_method(
+        only_broadcastable_along="space", new_dimension_name="normal"
+    )
+    def normal(self, on_same_side_as: ArrayLike = (0.0, 0.0)) -> np.ndarray:
+        """Compute the unit normal to this line.
+
+        The unit normal is a vector perpendicular to the input line
+        whose norm is equal to 1. The direction of the normal vector
+        is not fully defined: the line divides the 2D plane in two
+        halves, and the normal could be pointing to either of the half-planes.
+        For example, a horizontal line divides the 2D plane in a
+        bottom and a top half-plane, and we can choose whether
+        the normal points "upwards" or "downwards". We use a sample
+        point to define the half-plane the normal vector points to.
+
+        If this is a multi-segment line, the method raises an error.
+
+        Parameters
+        ----------
+        on_same_side_as : ArrayLike
+            A sample point in the (x,y) plane the normal is in. If multiple
+            points are given, one normal vector is returned for each point
+            given. By default, the origin is used.
+
+        Raises
+        ------
+        ValueError : When the normal is requested for a multi-segment geometry.
+
+        """
+        # A multi-segment geometry always has at least 3 coordinates.
+        if len(self.coords) > 2:
+            raise ValueError(
+                "Normal is not defined for multi-segment geometries."
+            )
+
+        on_same_side_as = np.array(on_same_side_as)
+
+        parallel_to_line = np.array(self.region.coords[1]) - np.array(
+            self.region.coords[0]
+        )
+        normal = np.array([parallel_to_line[1], -parallel_to_line[0]])
+        normal /= np.sqrt(np.sum(normal**2))
+
+        if np.dot((on_same_side_as - self.region.coords[0]), normal) < 0:
+            normal *= -1.0
+        return normal
+
+    def compute_angle_to_plane_normal(
+        self,
+        direction: xr.DataArray,
+        position: xr.DataArray,
+        angle_rotates: Literal[
+            "direction to normal", "normal to direction"
+        ] = "normal to direction",
+        in_degrees: bool = False,
+    ) -> xr.DataArray:
+        """Compute the angle between the normal to the segment and a direction.
+
+        This method is identical to
+        ``compute_egocentric_angle_to_nearest_point``, except that
+        rather than the angle between the approach vector and ``direction``,
+        the angle between the normal directed toward the segment and
+        ``direction`` is returned.
```

Comment:
```suggestion
```
I think we could do away with this bit (the part on "the angle between the normal directed toward the segment" is a bit confusing for example)

Reply:
Yeah, I don't mind this being scrapped.